bundles/org.openhab.binding.playstation/src/main/java/org/openhab/binding/playstation/internal/PS4Handler.java

```diff
@@ -0,0 +1,821 @@
+                break;
+            case CHANNEL_APPLICATION_IMAGE:
+                updateApplicationTitleid(currentApplicationId);
+                updateState(channelUID, currentArtwork);
+                break;
+            case CHANNEL_OSK_TEXT:
+            case CHANNEL_2ND_SCREEN:
+                updateState(channelUID, StringType.valueOf(""));
+                break;
+            case CHANNEL_KEY_UP:
+            case CHANNEL_KEY_DOWN:
+            case CHANNEL_KEY_RIGHT:
+            case CHANNEL_KEY_LEFT:
+            case CHANNEL_KEY_ENTER:
+            case CHANNEL_KEY_BACK:
+            case CHANNEL_KEY_OPTION:
+            case CHANNEL_KEY_PS:
+            case CHANNEL_DISCONNECT:
+                updateState(channelUID, OnOffType.OFF);
+                break;
+            default:
+                logger.warn("Channel refresh for {} not implemented!", channelUID.getId());
+        }
+    }
+
+    private void updateAllChannels() {
+        try (DatagramSocket socket = new DatagramSocket()) {
+            socket.setBroadcast(false);
+            socket.setSoTimeout(SOCKET_TIMEOUT_SECONDS * 1000);
+            InetAddress inetAddress = InetAddress.getByName(config.ipAddress);
+
+            // send discover
+            byte[] discover = PS4PacketHandler.makeSearchPacket();
+            DatagramPacket packet = new DatagramPacket(discover, discover.length, inetAddress, DEFAULT_BROADCAST_PORT);
+            socket.send(packet);
+
+            // wait for response
+            byte[] rxbuf = new byte[256];
+            packet = new DatagramPacket(rxbuf, rxbuf.length);
+            socket.receive(packet);
+            parseSearchResponse(packet);
+        } catch (IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+            logger.info("Fetch status exception: {}", e.getMessage());
+        }
+    }
+
+    private void stopConnection() {
+        SocketChannelHandler handler = socketChannelHandler;
+        if (handler != null && handler.isChannelOpen()) {
+            sendByeBye();
+        }
+    }
+
+    private void wakeUpPS4() {
+        logger.debug("Waking up PS4...");
+        try (DatagramSocket socket = new DatagramSocket()) {
+            socket.setBroadcast(false);
+            InetAddress inetAddress = InetAddress.getByName(config.ipAddress);
+            // send wake-up
+            byte[] wakeup = PS4PacketHandler.makeWakeupPacket(config.userCredential);
+            DatagramPacket packet = new DatagramPacket(wakeup, wakeup.length, inetAddress, DEFAULT_BROADCAST_PORT);
+            socket.send(packet);
+        } catch (IOException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+            logger.info("Wake up PS4 exception: {}", e.getMessage());
+        }
+    }
+
+    private class SocketChannelHandler extends Thread {
```

Comment:
Please move this to an own class, it's too long to be an inner class. Second: Why do you need this to be a seperate thread?

Reply:
My idea was to have all the communication in it's own class, that's why it's too long. Moved out the unnecessary methods for now.

Reply:
If I see correctly, you only send commands via this connection, but don't listen for data continously. So, this could be a scheduler task.